### PR TITLE
Rebase the new bootstrap commits on the version repositories

### PR DESCRIPTION
**Pull request by zimmski**
This PR redos the logic of updating a version repository. An update with the old logic looks like this https://github.com/go-clang/v3.4/pull/1 and the new one looks like this https://github.com/go-clang/v3.4/pull/2 Basically, instead of throwing away all old commits, we keep them and rebase bootstrap onto the version repository. This results into cleaner commits (no merges) and we can also see the real differences to the old revision of a version repository, instead of just the changes to the bootstrap repository.

The result can be seen here https://github.com/go-clang/v3.4/pull/2 https://github.com/go-clang/v3.6/pull/1 and https://github.com/go-clang/v3.7/pull/1

The result is much nicer since we can see what has changed with each commit.

### CODE REVIEW

**Comment by marxriedl** [comment]
Could you create an issue for this, to explain what this will solve/do?

**Comment by zimmski** [comment]
I'll adapt the PR description.

File: scripts/update-clang-version.sh [hunk 1]
```diff
@@ -1,24 +1,22 @@
 #!/bin/bash
 
+set -exuo pipefail
```
Comment (marxriedl) [inline scripts/update-clang-version.sh]: huh, didn't knew that, will incorporate that in my scripts

**Comment by marxriedl** [comment]
Nice, I like it.
One question though: doesn't this assume that in between nothing was pushed on e.g., 3.4.
The previous script (as far as i understood it) simply threw away those commits (not that that is a better handling)

File: scripts/update-clang-version.sh (hunk 1 above)
Comment (zimmski) [inline scripts/update-clang-version.sh]: There are even more flags (see "man set") I do not know why I forgot to use that.

**Comment by zimmski** [comment]
Yes, all previous commits were just dropped but recreated through the rebase. Every time!

**Pull request merged**